pkg/topology/kademlia/kademlia.go

```diff
@@ -1203,16 +1203,26 @@ func (k *Kad) EachPeerRev(f topology.EachPeerFunc) error {
 // onlyReachable wraps the iterator func to filer-out unreachable peers.
 func (k *Kad) onlyReachable(f topology.EachPeerFunc) topology.EachPeerFunc {
 	return func(addr swarm.Address, po uint8) (bool, bool, error) {
-		if !k.collector.Inspect(addr).IsReachable {
+		if k.collector.Inspect(addr).ReachabilityStatus != im.PeerReachabilityStatusPublic {
 			return false, false, nil
 		}
 		return f(addr, po)
 	}
 }
 
 // SetPeerReachability sets the peer reachability status.
-func (k *Kad) SetPeerReachability(addr swarm.Address, isReachable bool) {
-	k.collector.Record(addr, im.PeerReachability(isReachable))
+func (k *Kad) SetPeerReachability(addr swarm.Address, status string) error {
```

Comment:
passing the translated type could simplify things here (signature does not need to return error)